src/configuration-store/memory.store.ts

```diff
@@ -20,6 +20,11 @@ export class MemoryStore<T> implements ISyncStore<T> {
     Object.entries(entries).forEach(([key, val]) => {
       this.store[key] = val;
     });
+    Object.keys(this.store).forEach((key) => {
+      if (typeof entries[key] === 'undefined') {
+        delete this.store[key];
+      }
+    });
```

Comment:
```suggestion
    this.store = { ...entries };
    });
```

Reply:
🤦‍♂️ I'm embarrassed I did not think of this. Got anchored on extending the existing code--thanks for pulling me out of that trap!